spec/models/defect_spec.rb

```diff
@@ -124,6 +124,20 @@
     end
   end
 
+  describe '#reference_number' do
+    it 'returns a formatted identifier code' do
+      defect = create(:defect, sequence_number: 42)
+      expect(defect.reference_number).to eq('NB-000-042')
+    end
+
+    it 'uses sequential identifiers' do
+      defects = 5.times.map { create(:defect) }
```

Comment:
You could continue to use factory bot here by using `create_list(:defect, 5)`

Reply:
Thanks, I didn't know about that function :)